solution/src/integrationTest/kotlin/com/cosmotech/solution/service/SolutionServiceIntegrationTest.kt

```diff
@@ -2122,6 +2130,107 @@ class SolutionServiceIntegrationTest : CsmTestBase() {
+
+    solutionApiService.createSolutionFile(
+        organizationSaved.id, solutionSaved.id, multipartFile, true, null)
+
+    val fetchedFile =
+        solutionApiService.getSolutionFile(organizationSaved.id, solutionSaved.id, fileName)
+    val expectedText = expectedFile.bufferedReader().use { it.readText() }
+    val retrievedText = fetchedFile.inputStream.bufferedReader().use { it.readText() }
+    assertEquals(expectedText, retrievedText)
+  }
+
+  @Test
+  fun `test listSolutionFiles`() {
+    every { getCurrentAuthenticatedRoles(any()) } returns listOf("Platform.Admin")
+
+    logger.info("should list all solution file")
+    val resourceTestFile = resourceLoader.getResource("classpath:/$fileName").file
+    val input = FileInputStream(resourceTestFile)
+    val multipartFile =
+        MockMultipartFile(
+            "file", resourceTestFile.getName(), "text/plain", IOUtils.toByteArray(input))
+
+    var solutionFiles = solutionApiService.listSolutionFiles(organizationSaved.id, solutionSaved.id)
+    assertTrue(solutionFiles.isEmpty())
+
+    solutionApiService.createSolutionFile(
+        organizationSaved.id, solutionSaved.id, multipartFile, true, null)
+
+    solutionFiles = solutionApiService.listSolutionFiles(organizationSaved.id, solutionSaved.id)
+    assertEquals(1, solutionFiles.size)
+  }
+
+  @Test
+  fun `test deleteSolutionFile`() {
+    logger.info("should delete a solution file")
+    val resourceTestFile = resourceLoader.getResource("classpath:/$fileName").file
+    val input = FileInputStream(resourceTestFile)
+    val multipartFile =
+        MockMultipartFile(
+            "file", resourceTestFile.getName(), "text/plain", IOUtils.toByteArray(input))
+
+    solutionApiService.createSolutionFile(
+        organizationSaved.id, solutionSaved.id, multipartFile, true, null)
+
+    solutionApiService.deleteSolutionFile(organizationSaved.id, solutionSaved.id, fileName)
+
+    val exception =
+        assertThrows<NoSuchKeyException> {
+          solutionApiService.getSolutionFile(organizationSaved.id, solutionSaved.id, fileName)
+        }
+
+    assertEquals("The specified key does not exist.", exception.awsErrorDetails().errorMessage())
+  }
+
+  @Test
+  fun `test deleteSolutionFiles`() {
+    logger.info("should delete all solution files")
+    val resourceTestFile = resourceLoader.getResource("classpath:/$fileName").file
+    val input = FileInputStream(resourceTestFile)
+    val multipartFile =
+        MockMultipartFile(
+            "file", resourceTestFile.getName(), "text/plain", IOUtils.toByteArray(input))
+
+    solutionApiService.createSolutionFile(
+        organizationSaved.id, solutionSaved.id, multipartFile, true, null)
+    solutionApiService.createSolutionFile(
+        organizationSaved.id, solutionSaved.id, multipartFile, true, null)
+
+    solutionApiService.deleteAllS3SolutionObjects(organizationSaved.id, solutionSaved)
```

Comment:
Why are we not calling the actual endpoint implementation (`deleteSolutionFiles`)? Then we can remove this method from the service interface.